src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +50,28 @@
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              Todo #{todo.id}
             </div>
-
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
```

Comment:
Consider adding an accessible label to the close button to improve accessibility. This can be done by using the `aria-label` attribute.